pkg/envoy/rds/new_response.go

```diff
@@ -21,12 +22,25 @@ func newResponse(catalog catalog.MeshCataloger, proxy *envoy.Proxy) (*xds_discov
 		return nil, err
 	}
 
-	resp := &xds_discovery.DiscoveryResponse{
-		TypeUrl: string(envoy.TypeRDS),
+	// Get Ingress inbound policies for the proxy
+	services, err := catalog.GetServicesFromEnvoyCertificate(proxy.GetCertificateCommonName())
+	if err != nil {
+		log.Error().Err(err).Msgf("Error looking up services for Envoy with serial number=%q", proxy.GetCertificateSerialNumber())
+		return nil, err
+	}
+	for _, svc := range services {
+		ingressInboundPolicies, err := catalog.GetIngressPoliciesForService(svc, proxyIdentity)
```

Comment:
Is the reasoning for putting the ingress logic here (vs in catalog/routes.go) that it is related to service rather than service account? I'm wondering if we should move the ingress logic into catalog/routes.go so that file can house all logic related to building routes. What do you think?

Reply:
The ingress policies are build from the services on the proxy rather than the proxy identity. If I moved this logic into `catalog.ListTrafficPoliciesForServiceAccount(proxyIdentity)` I would have to modify the function and I didn't want to do that, hence I decided to keep this logic separate.

Another reason is that policies come from different sources : traffic targets, ingress we want to have these separated and combine them just before we configure the envoy. Putting this logic in the `newResponse` will help identify the different sources of traffic policies

Reply:
To be mindful of not tying outbound policies to services, I suggest not mixing ingress with the `ListTrafficPoliciesForServiceAccount()` API right now because this API currently deals with outbound routes as well and we do not want to bring in a dependency to specify the service there.

In the future when this API is split up into inbound and outbound, we can consider merging ingress routes within catalog for the inbound routes API.